.buildkite/docker-build-push.nix

```diff
@@ -0,0 +1,90 @@
+# This script will load nix-built docker images of cardano-node
+# into the Docker daemon (must be running), and then push to the
+# Docker Hub. Credentials for the hub must already be installed with
+# "docker login".
+#
+# There is a little bit of bash logic to replace the default repo and
+# tag from the nix-build (../nix/docker.nix).
+#
+# 1. The repo (default "inputoutput/cardano-node") is changed to match
+#    the logged in Docker user's credentials.
+#
+# 2. The tag (default "VERSION") is changed to reflect the
+#    branch which is being built under this Buildkite pipeline.
+#
+#    - If this is a git tag build (i.e. release) then the docker tag
+#      is left as-is.
```

Comment:
I don't think this description is valid for the implementation here. The build is first tagged with the commit hash, and then the build tagged with the possible git tags and the master branch tag. We're also not tagging `latest`